src/test_l2_dsDisplay.c

```diff
@@ -1,121 +1,172 @@
+        UT_LOG("Invoked dsGetDisplay() with dsVIDEOPORT_TYPE_HDMI and index 0. Return status: %d, Handle: %ld", ret, handle);
+        UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+        if (ret == dsERR_NONE)
+        {
+            // Step 3
+            ret = dsGetEDID(handle, edid);
+            UT_LOG("Invoked dsGetEDID() with handle. Return status: %d", ret);
+            UT_LOG("Parameter values ; productCode : %d , serialNumber :%d, manufactureYear : %d, manufactureWeek : %d, hdmiDeviceType : %d, isRepeater : %d, physicalAddressA :%d, physicalAddressB : %d, physicalAddressC : %d, physicalAddressD : %d, numOfSupportedResolution : %d, monitorName :%s ", edid->productCode, edid->serialNumber, edid->manufactureYear, edid->manufactureWeek, edid->hdmiDeviceType, edid->isRepeater, edid->physicalAddressA, edid->physicalAddressB, edid->physicalAddressC, edid->physicalAddressD, edid->numOfSupportedResolution, edid->monitorName );
+            UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+            // Step 4
+            // Validate the obtained EDID information with the predefined EDID value.
+            // This step is not implemented as it requires predefined EDID value.
+
+            // Step 5
+            ret = dsGetEDIDBytes(handle, edidBuffer, &length);
+            UT_LOG("Invoked dsGetEDIDBytes() with handle. Return status: %d, EDID Buffer Length: %d", ret, length);
+            UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+            // Step 6
+            // Validate the obtained EDID buffer with the predefined EDID buffer.
+            // This step is not implemented as it requires predefined EDID buffer.
+        }
+
+        // Step 7
+        ret = dsDisplayTerm();
+        UT_LOG("Invoked dsDisplayTerm(). Return status: %d", ret);
+        UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    }
+
+    UT_LOG("Exiting test_l2_dsDisplay_ValidateEDID...");
+}
 
 /**
-* @brief TODO: Describe the object of the test
+* @brief This test checks the default aspect ratio of the display
 *
-* TODO: Add the description of what is tested and why in this test
+* This test initializes the display, gets the display handle for HDMI port, checks the aspect ratio of the display and then terminates the display. The purpose of this test is to ensure that the default aspect ratio of the display is 16x9.
 *
-* **Test Group ID:** TODO: Add the group this test belongs to - Basic (for L1): 01 / Module (L2): 02 / Stress (L2): 03)@n
-* **Test Case ID:** TODO: Add the ID of the test case so that it can be logically tracked in the logs@n
+* **Test Group ID:** 02@n
+* **Test Case ID:** 002@n
 *
 * **Test Procedure:**
-* TODO: Refer to L2 Specification documentation : [l2_module_test_specification_template.md](../docs/pages/l2_module_test_specification_template.md)
+* Refer to UT specification documentation [l2_dsDisplay_test_specification.md](l2_dsDisplay_test_specification.md)
 */
-void test_l2_dsDisplay (void)
+
+void test_l2_dsDisplay_TestDefaultAspectRatio(void)
```

Comment:
as per the spec test_l2_dsDisplay_ValidateEDID is only for sink devices and test_l2_dsDisplay_TestDefaultAspectRatio is only for source devices. But there is no way we can differentiate. Its better to have some postfix (_sink, _source) to differentiate

Reply:
Done